content/english/java-tictactoe/activity-5.md

```diff
@@ -3,5 +3,80 @@ title: "5. (Optional) AI Move"
 description: ""
 date: 2020-07-23T00:00:00Z
 weight: 6
-hidden: true
----
\ No newline at end of file
+hidden: false
+---
+{{% notice tip %}}
+#### This is a bonus activity!
+To continue, you should have completed the Tic-Tac-Toe program from the previous exercises! 
+{{% /notice %}}
+
+### Let's make the computer think
+Previously, we use a `Random` object to randomly generate a move for the computer, making the competitiveness of the computer is pretty low.
+
+In this exercise, we want to increase the difficulty of the game by making optimal decisions for the computer.
+
+We will do so by adding artificial intelligence into our program with the use the <b>Minimax</b> algorithm (a well-defined procedure that allows computers to solve problems).
+
+### Minimax Algorithm
+Minimax is an algorithm that is often used in two-player games to make optimal decisions for a player. 
+
+- The algorithm makes optimal decision for the maximizer by examining all the possible future game states based on the current game board. 
+- If there is a winner or tie, the game state will be given a score. (Score: computer wins (`positive value`), player wins (`negative value`), tie (`0`)).
+- The two players are labeled as <b>maximizer</b> and <b>minimizer</b> respectedly. In this case, the computer is the maximizer, and the player is the minimizer. 
+- The maximizer tries to get the highest score possible, and the minimizer tries to minimize the score of its opponent. 
+
+<img src="../images/minimax.png" height="500"/> 
```

Comment:
Love the picture, great job! 💯 

Consider a few modifications:
1. mark/separate the rows to clearly show they belong to different rounds so that students know they need to go top down (like 1st row: start, 2nd row and so on)
2. Maybe calculate scores for each of the moves & mark the ones that should be chosen that round? Not sure if that if that will make the picture crowded  - i'll leave it upto you to decide what to do about this

Reply:
I change the arrows to make it clearer that each row is the different moves on can make! Thanks